nuxeo-runtime/nuxeo-connect-standalone/src/test/java/org/nuxeo/launcher/connect/TestConnectBroker.java

```diff
@@ -179,6 +192,62 @@ public void afterEach() {
         System.clearProperty(TomcatConfigurator.TOMCAT_HOME);
     }
 
+    // NXP-24507
+    @Test
+    public void testInstallPackageRequest_restartLauncherWithNoPending() {
+        // Given a downloaded package, which requires launcher restart
+        checkPackagesState(connectBroker, Arrays.asList("NXP-24507-A-1.0.0"), PackageState.DOWNLOADED);
+
+        // When handling the install request
+        try {
+            connectBroker.pkgRequest(null, Arrays.asList("NXP-24507-A-1.0.0"), null, null, true, false);
+            fail();
+        } catch (LauncherRestartException e) {
+            // Then an exception is raised
+        }
+        // And no file is created for pending changes (0 remaining).
+        Path pending = connectBroker.getInstallAfterRestartPath();
+        assertThat(pending).doesNotExist();
+    }
+
+    // NXP-24507
+    @Test
+    public void testInstallPackageRequest_restartLauncherWithOnePending() {
+        // Given a downloaded package, which requires launcher restart
+        String pkgA = "NXP-24507-A-1.0.0";
+        String pkgB = "NXP-24507-B-1.0.0";
+        checkPackagesState(connectBroker, Arrays.asList(pkgA, pkgB), PackageState.DOWNLOADED);
+
+        // When handling the install request
+        try {
+            connectBroker.pkgRequest(null, Arrays.asList(pkgA, pkgB), null, null, true, false);
+            fail();
+        } catch (LauncherRestartException e) {
+            // Then an exception is raised
+        }
+        // And a file is created for pending changes
+        Path pending = connectBroker.getInstallAfterRestartPath();
+        assertThat(pending).hasContent("install NXP-24507-B-1.0.0");
+    }
+
+    @Test
+    public void tesPersistPendingCommands_nonexistentPath() throws Exception {
+        Path path = connectBroker.getInstallAfterRestartPath();
+        assertThat(path).doesNotExist();
```

Comment:
?

Reply:
I'm testing the file creation here: I need to prove that the pending file is nonexistent first. (the tests are now renamed and comments are added to make it more explicit)